### PR TITLE
repair   shard key  underfined  bug

### PR DESCRIPTION
**Pull request by tomdong2010**
gastokenid add
transferTokenId add
shard key add

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -244,7 +244,14 @@ export default {
             fromEthAddress = web3in.eth.accounts[0];
           }
           if (obj.fromFullShardKey === undefined || obj.toFullShardKey === undefined) {
```
Comment (ping-ke) [inline index.js:246]: which on is the right one? fromFullShardId or fromFullShardKey?

File: index.js [hunk 2]
```diff
@@ -244,7 +244,14 @@ export default {
...
-            throw new Error('`fromFullShardKey` and `toFullShardKey` are required');
+            if (obj.fromFullShardId&&obj.fromFullShardId.length>0){
```
Comment (pingke) [inline index.js]: which on is the right one? fromFullShardId or fromFullShardKey?

Comment (ping-ke) [inline index.js]: which on is the right one? fromFullShardId or fromFullShardKey?

Comment (tomdong2010) [reply index.js]: fromFullShardId  is  in readme  .If not modify readme,use above,and I just committed it again

Comment (tomdong2010) [reply index.js]: > which on is the right one? fromFullShardId or fromFullShardKey?
ok ,  i have just repaired  it

File: index.js [hunk 3]
```diff
@@ -273,6 +274,9 @@ export default {
           //    '0x0' RLP-encoded transaction of all fields in Transaction (minus version, v, r, s)
           //    '0x1' typed encoding matching MetaMask initial implementation of EIP-712
           rawTx.version = '0x1';
+          rawTx.gasTokenId="0x8bb0";
```
Comment (ninjaahhh) [inline index.js]: this is not needed because of line 251

**Review by ninjaahhh — COMMENTED** [review]
seems this PR is not needed because token ID is already added

File: index.js (hunk 3 above)
Comment (tomdong2010) [reply index.js]: ok

Comment (tomdong2010) [reply index.js]: but  in  npm  ,  it  is not used  
image

File: index.js [hunk 4]
```diff
@@ -273,6 +274,8 @@ export default {
           //    '0x0' RLP-encoded transaction of all fields in Transaction (minus version, v, r, s)
           //    '0x1' typed encoding matching MetaMask initial implementation of EIP-712
           rawTx.version = '0x1';
+          rawTx.transferTokenId="0x8bb0";
```
Comment (ninjaahhh) [inline index.js]: also not needed because of line 251

Comment (tomdong2010) [reply index.js]: image

Comment (tomdong2010) [reply index.js]: but in npm , it is not used

File: index.js (hunk 3 above)
Comment (ninjaahhh) [reply index.js]: update the version to 2.0.0
https://cdn.jsdelivr.net/npm/quarkchain-web3@2.0.0
https://www.npmjs.com/package/quarkchain-web3

File: index.js (hunk 4 above)
Comment (tomdong2010) [reply index.js]: this is a  question, should  update npm , add "const rawTx = Object.assign({}, defaultTokenSetting, obj);"  and  "defaultTokenSetting"

**Pull request closed without merging**